internal/controller/reconciler.go

```diff
@@ -497,10 +504,47 @@ func deploymentForTunnel(r GenericTunnelReconciler) *appsv1.Deployment {
 							Requests: corev1.ResourceList{"memory": resource.MustParse("30Mi"), "cpu": resource.MustParse("10m")},
 							Limits:   corev1.ResourceList{"memory": resource.MustParse("256Mi"), "cpu": resource.MustParse("500m")},
 						},
+						SecurityContext: &corev1.SecurityContext{
+							AllowPrivilegeEscalation: pointer.To(false),
+							ReadOnlyRootFilesystem:   pointer.To(true),
+							RunAsUser:                pointer.To(int64(1002)),
```

Comment:
Any reason for this specific user? LGTM otherwise

Reply:
The Cloudflare container uses a non-numeric user in its Dockerfile, so Kubernetes with `runAsNonRoot` enabled cannot verify whether the user is root, resulting in an error.

Reply:
Ah I see, thanks. Looks good.